### PR TITLE
Documenting different hardware configuration aspects

### PR DESCRIPTION
**Pull request by rmohr**
* virtualization features
 * SMBIOS
 * Clock
 * Timer

Signed-off-by: Roman Mohr <rmohr@redhat.com>

### CODE REVIEW

File: machine.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+In order to provide a consistent view on the virtualized hardware for the Guest
+OS, the SMBIOS UUID can be set to a constant value via `spec.firmware.uid`:
+
+```
+metadata:
+  name: myvm
+spec:
+  domain:
+    firmware:
```
Comment (fabiand) [inline machine.md]: Didn't we say that `firmware` was to generic, and we wanted to go with:

```yaml
Domain:
  Loader:
    Firmware: uefi
    UUID: nnn-nnn-nnnn.....
# OR
  Bootloader:
    Smbios:
      UUID: nnnn-nnnn-nnnn-nnnnn
    Uefi|bios:
      Firmware: some reference to a blob
```

File: machine.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+      - name: myimage
+        volumeName: myimage
+        disk: {}
+  volumes:
+    - name: myimage
+      persistentVolumeClaim:
+        claimName: myclaim
+```
+
+**Note**: This is not related to scheduling decisions or resource assignment.
```
Comment (fabiand) [inline machine.md]: You might want to move this up

File: machine.md (hunk 1 above)
Comment (rmohr) [reply machine.md]: yes, but it is not yet implemented. Just documenting what is there.

File: machine.md (hunk 2 above)
Comment (rmohr) [reply machine.md]: ack

File: machine.md (hunk 1 above)
Comment (fabiand) [reply machine.md]: Ack

**Pull request merged**